client-app/shared/catalog/composables/useConfigurableProduct.ts

```diff
@@ -55,19 +61,24 @@ export function useConfigurableProduct(configurableProductId: string) {
   });
 
   async function fetchProductConfiguration() {
+    reset();
     fetching.value = true;
     try {
-      configuration.value = await getProductConfiguration(configurableProductId);
-      selectedConfigurationInput.value =
-        configuration.value?.configurationSections?.map((section) => ({
-          sectionId: section.id ?? "",
-          value: section.isRequired
-            ? {
-                productId: section.products?.[0]?.id ?? "",
-                quantity: section.quantity ?? 1,
-              }
-            : undefined,
-        })) ?? [];
+      const data = await getProductConfiguration(configurableProductId);
+      configuration.value = (data?.configurationSections as ConfigurationSectionType[]) ?? [];
+      configuration.value.forEach((section) => {
+        if (section.isRequired && section.id) {
+          changeSelectionValue({
+            sectionId: section.id,
+            value: {
+              productId: section.options?.[0]?.product?.id ?? "",
+              quantity: section.options?.[0]?.quantity ?? 1,
+            },
+          });
+        }
+      });
+      await nextTick();
```

Comment:
dirty hack?

Reply:
just forgot to remove that after some experiments :) thanks for pointing this out